DataManagementSystem/Client/ConsistencyInspector.py

```diff
@@ -235,6 +236,29 @@ def _getTSFiles( self ):
 
     return processedLFNs, nonProcessedLFNs, nonProcessedStatuses
 
+  def __getDirectories( self ):
+    """ get the directories where to look into (they are either given, or taken from the transformation ID
+    """
+    if self.directories:
+      directories = []
+      printout = False
+      for directory in self.directories:
+        if not directory.endswith( '...' ):
+          directories.append( directory )
+        else:
+          printout = True
+          topDir = os.path.dirname( directory )
+          res = self.fc.listDirectory( topDir )
+          if not res['OK']:
+            raise RuntimeError( res['Message'] )
+          else:
+            matchDir = directory.split( '...' )[0]
+            directories += [d for d in res['Value']['Successful'].get( topDir, {} ).get( 'SubDirs', [] ) if d.startswith( matchDir )]
+      if printout:
+        gLogger.always( 'Expanded list of %d directories:\n%s' % ( len( directories ), '\n'.join( directories ) ) )
+      return directories
+    else:                                                     
+      raise RuntimeError( "Need to specify the directories" )
```

Comment:
In DIRAC we do not use Exceptions except some special cases. Either return S_OK/S_ERROR(DError) or just return an empty list.